src/transformers/trainer.py

```diff
@@ -166,6 +166,8 @@ def __init__(
             # Set an xla_device flag on the model's config.
             # We'll find a more elegant and not need to do this in the future.
             self.model.config.xla_device = True
+        self.global_step = 0
+        self.epoch = 0
```

Comment:
We keep track of `global_step` and `epoch` so they can be accessible by `evaluate` method which is called at the end of the scripts.